include/shell/shell.h

```diff
@@ -396,6 +412,52 @@ struct shell_static_entry {
 #define SHELL_EXPR_CMD(_expr, _syntax, _subcmd, _help, _handler) \
 	SHELL_EXPR_CMD_ARG(_expr, _syntax, _subcmd, _help, _handler, 0, 0)
 
+/* Internal macro used for creating handlers for dictionary commands. */
+#define SHELL_CMD_DICT_HANDLER_CREATE(_data_, _handler_)		\
+static int UTIL_CAT(cmd_dict_, GET_ARG_N(1, __DEBRACKET _data_))(	\
+		const struct shell *shell, size_t argc, char **argv)	\
+{									\
+	return _handler_(shell, argc, argv,				\
+			 (void *)GET_ARG_N(2, __DEBRACKET _data_));	\
+}
+
+/* Internal macro used for creating dictionary commands. */
+#define SHELL_CMD_DICT_CREATE(_data_)					\
+	SHELL_CMD_ARG(GET_ARG_N(1, __DEBRACKET _data_), NULL, NULL,	\
+		UTIL_CAT(cmd_dict_, GET_ARG_N(1, __DEBRACKET _data_)), 1, 0)
+
+/**
+ * @brief Initializes shell dictionary commands.
+ *
```

Comment:
Please elaborate here when dictionary commands can be used.

Reply:
I am working on documentation now. Once I will have it I will update the description as well.